### ISSUE TITLE
mean_squared_error() the position of reduce_mean

### ISSUE THREAD

**Issue by Lucklyric** [zsdonghao/tensorlayer#209]
```python
def mean_squared_error(output, target, is_mean=False):
      with tf.name_scope("mean_squared_error_loss"):
        if output.get_shape().ndims == 2:   # [batch_size, n_feature]
            if is_mean:
                mse = tf.reduce_mean(tf.reduce_mean(tf.squared_difference(output, target), 1))
            else:
                mse = tf.reduce_mean(tf.reduce_sum(tf.squared_difference(output, target), 1))
        elif output.get_shape().ndims == 4: # [batch_size, w, h, c]
            if is_mean:
                mse = tf.reduce_mean(tf.reduce_mean(tf.squared_difference(output, target), [1, 2, 3]))
            else:
                mse = tf.reduce_mean(tf.reduce_sum(tf.squared_difference(output, target), [1, 2, 3]))
        return mse
```
The `mse `is acquired by first use `tf.reduce_sum`/`tf.reduce_mean` over feature space then use `tf.readuce_mean` to compute the mean over all the data points in the batch. It means that `is_mean` decides the loss is the sum or mean over feature space, not data space. 
However, in comments, it mentions `is_mean `  is used to decide the loss is from one data or otherwise. 

The confusion might be caused by my miss understanding of the comments. If it is not an issue for others, sorry about that, and please just close this issue.
